src/cdk/v2/destinations/movable_ink/procWorkflow.yaml

```diff
@@ -0,0 +1,71 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+  - name: toUnixTimestampInMS
+    path: ../../../../v0/util
+  - name: base64Convertor
+    path: ../../../../v0/util
+  - path: ./utils
+
+steps:
+  - name: messageType
+    template: |
+      .message.type.toLowerCase();
+
+  - name: validateInput
+    template: |
+      let messageType = $.outputs.messageType;
+      $.assert(messageType, "message Type is not present. Aborting");
+      $.assert(messageType in {{$.EventType.([.IDENTIFY,.TRACK])}}, "message type " + messageType + " is not supported");
+      $.assertConfig(.destination.Config.endpoint, "Movable Ink Endpoint is not present. Aborting");
+      $.assertConfig(.destination.Config.accessKey, "Access key is not present . Aborting");
+      $.assertConfig(.destination.Config.accessSecret, "Access Secret is not present. Aborting");
+      $.assert(.message.timestamp ?? .message.originalTimestamp, "Timestamp is not present. Aborting");
+
+      const userId = .message.().(
+          {{{{$.getGenericPaths("userIdOnly")}}}};
+      );
+      const email = .message.().(
+          {{{{$.getGenericPaths("email")}}}};
+      );
+
+      $.assert(userId ?? email ?? .message.anonymousId, "Either one of userId or email or anonymousId is required. Aborting");
+
+  - name: preparePayload
+    template: |
+      const userId = .message.().(
+          {{{{$.getGenericPaths("userIdOnly")}}}};
+      );
+      const email = .message.().(
+          {{{{$.getGenericPaths("email")}}}};
+      ); 
+      const timestampInUnix = $.toUnixTimestampInMS(.message.().(
+            {{{{$.getGenericPaths("timestamp")}}}};
+      ));
+      $.context.payload = {
```

Comment:
As I gather from research spec, movable ink supports most of the rudderstack standard ecomm events. And I see that we have documented the by default mapping in the research spec as well. But in implementation, I do not see any mapping.json file. Which means we are actually allowing whatever fields coming as input, straight as output event. What is the thought behind this? 
Is it just because that the user needs to configure every single field inside the movable, that is why we are giving such liberty ?
But in the spec, I see we have multiple required field. Where is the check for handling that if missing?

Reply:
In movable ink we have to configure the each event schema, So, we don't have a way to know if user is configured the schema in movable ink or not. Thats why we are passing the event as it is.
It is more like a documented heavy destination than the implementation wise.

Reply:
Which means, even if some fields are stated as mandatory, if we do not send, it will not throw error?

Reply:
Yes